lib/Widget/ForecastIo.php

```diff
@@ -502,29 +577,44 @@ public function getResource($displayId = 0)
         }
 
         // Handle the daily template (if its here)
-        if (stripos($body, '[dailyForecast]')) {
+        $dailySubs = '';
+        $matches = '';
+        preg_match_all('/\[dailyForecast.*?\]/', $body, $matches);
+        // Substitute
+        foreach ($matches[0] as $sub) {
+            $replace = str_replace('[', '', str_replace(']', '', $sub));
+            // Handling for date/time
+            $itterations = 7;
+            $offset = 0;
+            if (stripos($replace, '|') > -1) {
+                $quantity = explode('|', $replace);
+                $itterations = $quantity[1];
+                
+                if (count($quantity) > 1)
+                  $offset = $quantity[2];
+            }
+
             // Pull it out, and run substitute over it for each day
-            $dailySubs = '';
             // Substitute for every day (i.e. 7 times).
-            for ($i = 0; $i < 7; $i++) {
+            for ($i = $offset; $i < $itterations + $offset; $i++) {
```

Comment:
I am not convinced this is correct - lets say you want all 7 days, starting day 2 - `$itterations = 7` and `$offset = 2`, that means the loop will go until `$i < 9` ?